libnest/src/cache/depgraph/graph.rs

```diff
@@ -452,24 +449,38 @@ impl<'lock_file> DependencyGraph<'lock_file> {
             .for_each(|node_id| self.remove_node(node_id));
     }
 
+    fn find_node_for_requirement(&self, requirement: &PackageRequirement) -> Option<NodeID> {
```

Comment:
I feel like this function's name is kinda bad, mostly because when reading your code, I thought, only by its name, that it would find *the node that fulfills the given requirement*, while it's not exactly that.